packages/components/src/header/index.tsx

```diff
@@ -0,0 +1,64 @@
+import {
+	__experimentalVStack as VStack,
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+	__experimentalHeading as Heading,
+} from '@wordpress/components';
+import { Breadcrumbs } from '../breadcrumbs';
+import type { HeaderProps } from './types';
+
+import './style.scss';
+
+/**
+ * The Header component provides a structured introduction to a page or section,
+ * combining a title, optional description, and contextual actions. It supports
+ * varying levels of hierarchy through semantic heading levels, and can include
+ * visual decorations, navigational aids like breadcrumbs, and utility controls
+ * such as buttons or dropdowns.
+ *
+ * ```jsx
+ * import { Header } from '@automattic/components';
+ * import { Button } from '@wordpress/components';
+ * import { cog } from '@wordpress/icons';
+ *
+ * function MyComponent() {
+ * 	return (
+ * 		<Header
+ * 			level={1}
+ * 			title="Settings"
+ * 			description="Configure your application settings"
+ * 			decoration={<Icon icon={cog} />}
+ * 			actions={<Button variant="primary">Save Changes</Button>}
+ * 		/>
+ * 	);
+ * }
+ * ```
+ */
+export const Header = ( {
```

Comment:
I noticed that recent DS components, e.g. `SummaryButton`, export this as `default`. Should we have some consistency here?

Reply:
My preference is for named exports over default exports:  they are less ambiguous and easier to trace across the codebase, and more tool-friendly (static analysis, tree-shaking, etc).

And whatever approach we choose, I agree that we could do a pass throughout the package to uniform things a bit